source/core/src/main/com/csse3200/game/components/inventory/PlayerInventoryDisplay.java

```diff
@@ -82,30 +86,96 @@ public PlayerInventoryDisplay(Inventory inventory, int numCols, int hotBarCapaci
     @Override
     public void create() {
         super.create();
+        dragAndDrop = new DragAndDrop();
         generateHotBar();
-        entity.getEvents().addListener("toggleInventory", this::toggleInventory);
+        entity.getEvents().addListener("toggleInventory", this::toggleDisplay);
         entity.getEvents().addListener("addItem", this::addItem);
     }
 
     /**
      * Toggles the inventory display on or off based on its current state.
      */
-    private void toggleInventory() {
-        if (stage.getActors().contains(inventoryDisplay, true)) {
+    private void toggleDisplay() {
+        toggle = !stage.getActors().contains(inventoryDisplay, true);
+        logger.info("TOGGLING!");
```

Comment:
Change the comment inside the log and change the debug level to debug rather than info